### PR TITLE
feat(freeMode): init

### PR DESCRIPTION
**Pull request by vltansky**
- [x] Should target `v7` branch
- [x] test `enabled: true`

#4228

### CODE REVIEW

File: src/components/core/classes/addClasses.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default function addClasses() {
     'initialized',
     params.direction,
     { 'pointer-events': support.pointerEvents && !support.touch },
-    { 'free-mode': params.freeMode },
+    { 'free-mode': params.freeMode.enabled },
```
Comment (nolimits4web) [inline src/components/core/classes/addClasses.js:25]: `params.freeMode && params.freeMode.enabled`

File: src/components/core/core-class.js [hunk 2]
```diff
@@ -331,7 +331,7 @@ class Swiper {
       swiper.updateSlidesClasses();
     }
     let translated;
-    if (swiper.params.freeMode) {
+    if (swiper.params.freeMode.enabled) {
```
Comment (nolimits4web) [inline src/components/core/core-class.js:334]: `swiper.params.freeMode && swiper.params.freeMode.enabled`

File: src/components/core/events/onTouchEnd.js [hunk 3]
```diff
@@ -74,178 +74,8 @@ export default function onTouchEnd(event) {
...
-      swiper.slideToClosest();
-      return;
-    }
-
-    if (!params.freeModeMomentum || timeDiff >= params.longSwipesMs) {
-      swiper.updateProgress();
-      swiper.updateActiveIndex();
-      swiper.updateSlidesClasses();
-    }
+  if (params.freeMode.enabled) {
```
Comment (nolimits4web) [inline src/components/core/events/onTouchEnd.js:77]: `params.freeMode && params.freeMode.enabled`

File: src/components/core/events/onTouchMove.js [hunk 4]
```diff
@@ -212,18 +212,8 @@ export default function onTouchMove(event) {
...
-      data.velocities.push({
-        position: touches[swiper.isHorizontal() ? 'startX' : 'startY'],
-        time: data.touchStartTime,
-      });
-    }
-    data.velocities.push({
-      position: touches[swiper.isHorizontal() ? 'currentX' : 'currentY'],
-      time: now(),
-    });
+  if (params.freeMode.enabled) {
```
Comment (nolimits4web) [inline src/components/core/events/onTouchMove.js:215]: `params.freeMode && params.freeMode.enabled`

**Review by nolimits4web — CHANGES_REQUESTED** [review]

**Comment by vltansky** [comment]
@nolimits4web can you create `v7` branch?

**Comment by vltansky** [comment]
**Size Change:** -3.12 kB (-1%) 

**Total Size:** 490 kB

| Filename | Size | Change |  |
| :--- | :---: | :---: | :---: |
| `./package/angular/bundles/swiper_angular.umd.js` | 12 kB | -75 B (-1%) |  |
| `./package/angular/bundles/swiper_angular.umd.min.js` | 7.32 kB | -66 B (-1%) |  |
| `./package/angular/esm2015/angular/src/swiper.component.js` | 22.4 kB | -152 B (-1%) |  |
| `./package/angular/esm2015/angular/src/utils/params-list.js` | 3.05 kB | -141 B (-4%) |  |
| `./package/angular/fesm2015/swiper_angular.js` | 7.6 kB | -91 B (-1%) |  |
| `./package/cjs/components/core/classes/addClasses.js` | 590 B | +15 B (+3%) |  |
| `./package/cjs/components/core/core-class.js` | 4.61 kB | +7 B (0%) |  |
| `./package/cjs/components/core/defaults.js` | 1.21 kB | -69 B (-5%) | ✅ |
| `./package/cjs/components/core/events/onTouchEnd.js` | 1.16 kB | -1.35 kB (-54%) | 🏆 |
| `./package/cjs/components/core/events/onTouchMove.js` | 1.86 kB | -65 B (-3%) |  |
| `./package/cjs/components/lazy/lazy.js` | 2.5 kB | +1 B (0%) |  |
| `./package/cjs/react/params-list.js` | 913 B | -42 B (-4%) |  |
| `./package/cjs/svelte/params-list.js` | 913 B | -42 B (-4%) |  |
| `./package/cjs/vue/params-list.js` | 913 B | -42 B (-4%) |  |
| `./package/cjs/vue/swiper.js` | 2.96 kB | -45 B (-1%) |  |
| `./package/esm/components/core/classes/addClasses.js` | 562 B | +16 B (+3%) |  |
| `./package/esm/components/core/core-class.js` | 4.48 kB | +7 B (0%) |  |
| `./package/esm/components/core/defaults.js` | 1.16 kB | -69 B (-6%) | ✅ |
| `./package/esm/components/core/events/onTouchEnd.js` | 1.13 kB | -1.35 kB (-54%) | 🏆 |
| `./package/esm/components/core/events/onTouchMove.js` | 1.76 kB | -68 B (-4%) |  |
| `./package/esm/components/mousewheel/mousewheel.js` | 4.46 kB | -1 B (0%) |  |
| `./package/esm/react/params-list.js` | 881 B | -44 B (-5%) | ✅ |
| `./package/esm/svelte/params-list.js` | 881 B | -44 B (-5%) | ✅ |
| `./package/esm/vue/params-list.js` | 881 B | -44 B (-5%) | ✅ |
| `./package/esm/vue/swiper.js` | 2.91 kB | -42 B (-1%) |  |
| `./package/swiper-bundle.cjs.js` | 979 B | +28 B (+3%) |  |
| `./package/swiper-bundle.esm.browser.js` | 56.2 kB | +166 B (0%) |  |
| `./package/swiper-bundle.esm.browser.min.js` | 38.9 kB | +121 B (0%) |  |
| `./package/swiper-bundle.esm.js` | 619 B | +19 B (+3%) |  |
| `./package/swiper-bundle.js` | 56.9 kB | +150 B (0%) |  |
| `./package/swiper-bundle.min.js` | 38 kB | +153 B (0%) |  |
| `./package/swiper.cjs.js` | 552 B | +18 B (+3%) |  |
| `./package/swiper.esm.js` | 523 B | +17 B (+3%) |  |

**Pull request merged**